test/widgets/lightbox_test.dart

```diff
@@ -197,6 +203,136 @@ class FakeVideoPlayerPlatform extends Fake
+      addTearDown(testBinding.reset);
+      final subscription = eg.subscription(channel);
+      await testBinding.globalStore.add(eg.selfAccount, eg.initialSnapshot(
+        streams: [channel], subscriptions: [subscription]));
+      store = await testBinding.globalStore.perAccount(eg.selfAccount.id);
+      connection = store.connection as FakeApiConnection;
+      await store.addUser(eg.selfUser);
+
+      connection.prepare(json:
+        eg.newestGetMessagesResult(foundOldest: true, messages: [message]).toJson());
+
+      await tester.pumpWidget(TestZulipApp(accountId: eg.selfAccount.id,
+        child: MessageListPage(initNarrow: const CombinedFeedNarrow())));
+
+      await tester.pumpAndSettle();
+    }
+
+    testWidgets('Hero animation occurs smoothly when opening lightbox from message list', (tester) async {
+      prepareBoringImageHttpClient();
+
+      await setupMessageListPage(tester);
+
+      final messageContentFinder = find.byWidgetPredicate((widget) =>
+        widget is MessageContent && widget.message.id == message.id
+      );
+      final messageListImageFinder = find.descendant(
+        of: messageContentFinder,
+        matching: find.byType(RealmContentNetworkImage)
+      );
+      final imageSrcUrl = tester
+        .widget<RealmContentNetworkImage>(messageListImageFinder).src;
+      final initialImagePosition = tester.getRect(messageListImageFinder);
+      await tester.tap(messageListImageFinder);
+      await tester.pump();
+      // pump to start hero animation
+      await tester.pump();
+
+      const heroAnimationDuration = Duration(milliseconds: 300);
+      const steps = 150;
+      final stepDuration = heroAnimationDuration ~/ steps;
+      List<Rect> animatedPositions = [];
+      for (int i = 1; i <= steps; i++) {
+        await tester.pump(stepDuration);
+
+        final animatedFlightImageFinder = find.byWidgetPredicate((widget) =>
+          widget is RealmContentNetworkImage && widget.src == imageSrcUrl
+        );
+        final animatedFlightImagePosition = tester.getRect(animatedFlightImageFinder);
+        animatedPositions.add(animatedFlightImagePosition);
+      }
+
+      final totalTopDistance = (
+        initialImagePosition.top - animatedPositions.last.top).abs();
+      final totalLeftDistance = (
+        initialImagePosition.left - animatedPositions.last.left).abs();
+
+      Rect previousPosition = initialImagePosition;
+      for (final position in animatedPositions) {
+        final positionChangeTop = position.top - previousPosition.top;
+        final positionChangeLeft = position.left - previousPosition.left;
+        check(positionChangeTop.abs()).isLessThan(0.1 * totalTopDistance);
+        check(positionChangeLeft.abs()).isLessThan(0.1 * totalLeftDistance);
+        check(position).not((pos) => pos.equals(previousPosition));
+
+        previousPosition = position;
+      }
+      await tester.pumpAndSettle();
```

Comment:
If this is needed, that suggests that the animation isn't actually over when this test stops checking.